hastory-server/src/Data/Hastory/Server.hs

```diff
@@ -98,21 +131,43 @@ generateToken = T.pack . take tokenLength . randomRs ('a', 'z') <$> liftIO newSt
 reportPort :: MonadLogger m => Options -> m ()
 reportPort Options {..} = logInfo $ "Starting server on port " <> T.pack (show _oPort)
 
--- | Logs information about the data file. This data file serves as a database for this primitive append-only server.
-reportDataFileStatus :: (MonadIO m, MonadLogger m) => Options -> m ()
-reportDataFileStatus Options {..} = do
-  dataFileExists <- liftIO $ fileExist _oDataOutputFilePath
-  if dataFileExists
-    then logInfo $
-         "Data file exists at " <> T.pack _oDataOutputFilePath <> ". Appending commands to it."
-    else logInfo "Data file doesn't exist. Creating a new one."
+-- | We use a hardcoded file name for database. Path to database is chosen by the user.
+mkDbFile :: FilePath -> FilePath
+mkDbFile dir = dir </> "hastory.db"
+
+-- | We use a hardcoded file name for file data storage. Path to database is chosen by the user.
+mkStorageFile :: FilePath -> FilePath
+mkStorageFile = (</> "data.txt")
+
+-- | Creates directory if it cannot be found. Only difference between this
+-- function and `System.Directory.createDirectoryIfMissingSource` is that
+-- this function performs logging.
+createDirectoryIfMissing :: (MonadIO m, MonadLogger m) => FilePath -> m ()
+createDirectoryIfMissing dir = do
+  directoryExists <- liftIO $ Dir.doesDirectoryExist dir
+  if directoryExists
+    then logInfo $ "Data directory already in " <> T.pack dir <> "/. Not changing anything."
+    else do
+      logInfo $ "Data directory doesn't exist. Creating a new one at " <> T.pack dir <> "/."
+      liftIO $ Dir.createDirectory dir
+
+-- | Logs information about the data storage.
+reportStorageStatus :: (MonadUnliftIO m, MonadLogger m) => Options -> m ()
+reportStorageStatus Options {..} =
+  case _oStorage of
+    FileBasedStorage -> createDirectoryIfMissing _oDataDirectory
+    SQLiteDatabase -> do
+      createDirectoryIfMissing _oDataDirectory
+      let dbFile = T.pack $ mkDbFile _oDataDirectory
+      DB.runSqlite dbFile $ DB.runMigration migrateAll
+      logInfo $ "Using " <> dbFile <> " as SQLite database. Data is synchronised."
```

Comment:
This function does more than just reporting and it's only used in one place.
If we'll get rid of the file-based storage anyway, I think we should inline it.